keps/sig-network/3015-node-level-topology/README.md

```diff
@@ -0,0 +1,502 @@
+that previously had standard `Cluster` semantics would now get
+node-level topology. Of course, we would only use the "Automatic"
+approach if we believed this was a safe change.
+
+When downgrading, older kube-controller-manager / kube-proxy would not
+be aware of node-level topology, and would just fall back to ordinary
+`Cluster` semantics for the affected services, resulting in
+less-efficient but still correct functioning.
+
+### Version Skew Strategy
+
+For the most part, version skew will result in node-level topology not
+happening and the service falling back to ordinary `Cluster`
+semantics.
+
+One tricky case is that if kube-controller-manager is downgraded (or
+has its feature gate disabled), but kube-apiserver and kube-proxy stay
+new/enabled, then a service might get "stuck" with node-level topology
+enabled in its EndpointSlices. Even this is unlikely to be a big
+problem; services are not likely to flip back and forth between
+"suitable for node-level topology" and "not suitable for node-level
+topology", so if the previous version of kcm declared the service to
+be suitable for node-level topology, then it probably doesn't matter
+if the new version doesn't keep checking that it's still suitable.
+(And even in the worse case, the result is just that the endpoints get
+used in an unbalanced way, not that the service actually breaks.) And
+the version skew should eventually get resolved.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: NodeLevelTopology
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kube-controller-manager
+    - kube-proxy
+
+###### Does enabling the feature change any default behavior?
+
+Potentially, depending on UNRESOLVED decisions above.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes; services that were using node-level topology will just revert
+back to ordinary `Cluster` semantics.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+It starts working again.
+
+###### Are there any tests for feature enablement/disablement?
+
+Not yet.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Assuming we use the "Automatic" semantics, a rollout would affect the
+behavior of existing services (such as CoreDNS) that met the criteria
+for automatically getting node-local topology. Assuming the feature
+works correctly though, it would affect them in a positive way.
+
+If we used the "Manual" semantics then rolling out the feature should
+have no effect at all.
+
+###### What specific metrics should inform a rollback?
+
+There are no metrics that would inform anyone that the feature was
```

Comment:
Increases in kube-proxy programming time? Increases in endpointslice controller reconciliation times? Basically, what metrics cover the parts of code you are touching, so if there is a bug it can be detected before users complain.